coordinator/src/db/funding_rates.rs

```diff
@@ -0,0 +1,89 @@
+use crate::schema::funding_rates;
+use anyhow::Context;
+use anyhow::Result;
+use diesel::prelude::*;
+use rust_decimal::prelude::FromPrimitive;
+use rust_decimal::prelude::ToPrimitive;
+use rust_decimal::Decimal;
+use time::OffsetDateTime;
+use xxi_node::commons::to_nearest_hour_in_the_past;
+
+#[derive(Queryable, Debug)]
+struct FundingRate {
+    #[diesel(column_name = "id")]
+    _id: i32,
+    start_date: OffsetDateTime,
+    end_date: OffsetDateTime,
+    rate: f32,
+    #[diesel(column_name = "timestamp")]
+    _timestamp: OffsetDateTime,
+}
+
+pub(crate) fn insert(
+    conn: &mut PgConnection,
+    funding_rates: &[xxi_node::commons::FundingRate],
+) -> Result<()> {
+    conn.transaction(|conn| {
+        for funding_rate in funding_rates {
```

Comment:
Correct me if I'm wrong @luckysori but I think we said this was to catch errors for a single insert and not fail the whole batch.

Reply:
Yeah, this was intentional.

Reply:
I don't get it, if the first insert fails it would be the same behaviour as if a batch insert fails. 

Also we only calling the database here. So if that fails I doubt the next insert will be working.